api/ably-token-request/index.ts

```diff
@@ -2,15 +2,21 @@ import "../startup";
 import { Context, HttpRequest } from "@azure/functions";
 import { authorized, ApiRequestContext } from "../common/ApiRequestContext";
 import * as Ably from "ably/promises";
-
-const client = new Ably.Rest(process.env.ABLY_API_KEY);
+import { UserService } from "../common/services/UserService";
 
 export default async function (context: Context, req: HttpRequest): Promise<void> {
   await authorized(
     context,
     req,
     async ({ user }: ApiRequestContext) => {
       const clientId = `${user.id}:${user.username}:${encodeURIComponent(user.profileImgUrl)}`;
+      
+      const userService = new UserService();
+      const { role } = await userService.getRoleByUsername(user.username);
+
+      const apiKey = role ? role.apiKey : process.env.ABLY_API_KEY;
```

Comment:
I think we'd need to make some decisions around that and whether this is the best way to fix the issue, basically it was a catch for if somehow a user has been created with no role assigned (like my testyjofrank account, which was created before the roles were added) we should be funnelling users into success rather than leaving them hanging if something weird (like no role assigned) has happened.

Happy for you to update this later if you think there's a better default option. Maybe we set up a specific API key

Reply:
My thinking would be to default any user without a role to a base role (`normal`?), so that they're then no longer without a role incorrectly. WDYT?

Reply:
Sounds good. Not sure about the name 'normal' perhaps 'user'? then we have user and admin as two kinda standard roles for the demo?